torch/optim/lr_scheduler.py

```diff
@@ -896,22 +902,36 @@ class ChainedScheduler(LRScheduler):
         >>> # lr = 0.59049  if epoch >= 4
         >>> scheduler1 = ConstantLR(optimizer, factor=0.1, total_iters=2)
         >>> scheduler2 = ExponentialLR(optimizer, gamma=0.9)
-        >>> scheduler = ChainedScheduler([scheduler1, scheduler2])
+        >>> scheduler = ChainedScheduler([scheduler1, scheduler2], optimizer=optimizer)
         >>> for epoch in range(100):
         >>>     train(...)
         >>>     validate(...)
         >>>     scheduler.step()
     """
 
-    def __init__(self, schedulers):
-        for scheduler_idx in range(1, len(schedulers)):
-            if (schedulers[scheduler_idx].optimizer != schedulers[0].optimizer):
+    def __init__(self, schedulers: Sequence[LRScheduler], optimizer: Optional[Optimizer] = None):
+        if len(schedulers) < 1:
+            raise ValueError("ChainedScheduler expects at least one schedulers to be chained, but got no scheduler.")
+
+        optimizer = optimizer or schedulers[0].optimizer
+        for sch_idx, scheduler in enumerate(schedulers):
+            if not isinstance(scheduler, LRScheduler):
+                raise TypeError(
+                    f"{self.__class__.__name__} expects all schedulers to be of type LRScheduler, "
+                    f"but got {type(scheduler)}"
+                )
+
+            if not sch_idx and optimizer is None:
+                optimizer = scheduler.optimizer
```

Comment:
Was this use case allowed before? Could people have passed in an optimizer to just one of the LRSchedulers in the chain and have the rest be empty? If not, this check would be unnecessary.

Reply:
Oh I see. this line is totally unnecessary as its already covered by L916. will remove this.